src/export/dashboard.ts

```diff
@@ -0,0 +1,58 @@
+// (C) 2019 GoodData Corporation
+import { DashboardExport } from "@gooddata/typings";
+import { ApiResponse, ApiResponseError, XhrModule } from "../xhr";
+import { IExportResponse } from "../interfaces";
+import { handleHeadPolling, IPollingOptions } from "../util";
+import { sanitizeDateFilters, isDataExported } from "./util";
+
+interface IExportDashboardPayload {
+    dashboardExport: {
+        dashboardUri: string;
+        filters?: DashboardExport.FilterContextItem[];
+    };
+}
+
+export class DashboardModule {
+    constructor(private xhr: XhrModule) {}
+
+    public exportDashboardToPdf(
+        projectId: string,
+        dashboardUri: string,
+        filters: DashboardExport.FilterContextItem[] = [],
```

Comment:
after discussion with @lupko it would make sense to use on gooddata-js API our desired interfaces which are https://github.com/gooddata/gooddata-typings/blob/master/src/VisualizationInput.ts so please use Filter definitions from this VisualizationInput namespace...

inside we can convert to `FilterContextItem` which are required by backend also consider to remove `number` type from `from, to` properties in `DateFilter` definition as they are not supported by backend. 

@lupko correct me if something is not right...

Note: take into account that this gooddata-js API is public so creating just followup to make it right will cause the issues with deprecation...

Reply:
As I discussed with you and @lupko , we misuse `VisualizationInput` here. It's only for every single widget, not whole dashboard. So I will keep my current implementation.